src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -397,7 +405,7 @@ def schema_query(self) -> str:
 
     def to_subqueryable(self) -> "SelectSQL":
         """Convert this SelectSQL to a new one that can be used as a subquery. Refer to __init__."""
-        if self.convert_to_select:
+        if self.convert_to_select or is_sql_select_statement(self._sql_query):
```

Comment:
do we really need it? the constructor of SelectSQL already checks it?

Reply:
Without this, select statements create a new `SelectSQL` and assign column_states = self.column_states.

In case when `self._sql_query` is `SELECT SYSTEM$WAIT(3)`, processing `column_states` triggers a describe query and it ends up in a blocking call. This was causing async test jobs failing.

After writing this explanation, I feel like we should only set `new.column_states = self.column_states` if `self.column_states is not None`

Reply:
yeah make sense